baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/AutoCloseableMustBeClosed.java

```diff
@@ -0,0 +1,83 @@
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.sun.source.tree.MethodTree;
+import java.util.stream.BaseStream;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "AutoCloseableMustBeClosed",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        providesFix = BugPattern.ProvidesFix.REQUIRES_HUMAN_ATTENTION,
+        severity = SeverityLevel.SUGGESTION,
+        summary = AutoCloseableMustBeClosed.SUMMARY)
+public final class AutoCloseableMustBeClosed extends BugChecker implements MethodTreeMatcher {
+
+    static final String SUMMARY = "If a constructor or method returns an AutoCloseable, "
+            + "it should be annotated @MustBeClosed to ensure callers appropriately close resources";
+    private static final String MUST_BE_CLOSED_TYPE = "com.google.errorprone.annotations.MustBeClosed";
+    private static final String CAN_IGNORE_RETURN_VALUE_TYPE = "com.google.errorprone.annotations.CanIgnoreReturnValue";
+
+    private static final Matcher<MethodTree> methodReturnsAutoCloseable = Matchers.allOf(
+            Matchers.not(Matchers.methodIsConstructor()),
+            Matchers.methodReturns(Matchers.isSubtypeOf(AutoCloseable.class)),
+            // ignore Stream for now, see https://errorprone.info/bugpattern/StreamResourceLeak
+            Matchers.not(Matchers.methodReturns(Matchers.isSubtypeOf(BaseStream.class))));
+
+    private static final Matcher<MethodTree> constructsAutoCloseable = Matchers.allOf(
+            Matchers.methodIsConstructor(),
+            Matchers.enclosingClass(Matchers.isSubtypeOf(AutoCloseable.class)),
+            // ignore Stream for now, see https://errorprone.info/bugpattern/StreamResourceLeak
+            Matchers.not(Matchers.enclosingClass(Matchers.isSubtypeOf(BaseStream.class))));
+
+    private static final Matcher<MethodTree> methodNotAnnotatedMustBeClosed =
+            Matchers.not(Matchers.hasAnnotation(MUST_BE_CLOSED_TYPE));
+
+    private static final Matcher<MethodTree> methodNotAnnotatedIgnoreReturnValue =
+            Matchers.not(Matchers.hasAnnotation(CAN_IGNORE_RETURN_VALUE_TYPE));
+
+    private static final Matcher<MethodTree> methodShouldBeAnnotatedMustBeClosed = Matchers.allOf(
+            Matchers.anyOf(methodReturnsAutoCloseable, constructsAutoCloseable),
+            methodNotAnnotatedMustBeClosed,
+            methodNotAnnotatedIgnoreReturnValue);
+
+    @Override
+    public Description matchMethod(MethodTree tree, VisitorState state) {
+        if (methodShouldBeAnnotatedMustBeClosed.matches(tree, state)) {
+            SuggestedFix.Builder builder = SuggestedFix.builder().addImport(MUST_BE_CLOSED_TYPE);
+            String annotation = SuggestedFixes.qualifyType(state, builder, MUST_BE_CLOSED_TYPE);
```

Comment:
`SuggestedFixes.qualifyType(state, builder, MUST_BE_CLOSED_TYPE)` actually adds the import for us, the `addImport` call can be removed.